### DISCUSSION TITLE
Você já chegou a trabalhar com back-end?

### DISCUSSION THREAD

**Discussion by erickod** [frontendbr/forum#1022] [Discussão]
Em se tratando do escopo de um projeto, após levantados o requisitos e com uma documentação elaborada, seria melhor começar a desenvolver o projeto pelo front-end ou pelo back-end, em sua opinião?

Aqui a equipe é bem reduzida.

**Comment by fredmosc** [comment]
Aqui na empresa nós trabalhamos com o back-end primeiro, apenas fazemos
algum prototipo de front-end para validacao com o  cliente

**Comment by ninetails** [comment]
O ideal é ter uma boa gerência (não necessariamente um gerente, mas digo do time), onde entre os integrantes será definido um roadmap junto com o pessoal de produto ou o cliente e ver o que agrega mais valor, no caso de entregas contínuas por exemplo.

Se for utilizado metodologias ágeis, dá pra fazer por partes entregáveis do domínio, algo como fazer o back de um entregável, depois o front, depois passar para outro projeto. Ou às vezes mudar isso também, começar pelo front se o back não estiver bem definido como especificação de produto. Ou backend antes se a especificação das telas não estiverem ok ou não tiver layout... por aí vai

**Comment by luizbills** [comment]
Eu trabalho solo. Não sei opinar isso. Mas acredito que dá para trabalho o front e o back paralelamente, ao meu ver nenhum influencia direto no outro a ponto de atrapalhar no desenvolvimento.

**Comment by giioohbernini** [comment]
Quando eu fazia freela com fonte de renda principal, costumava fazer o
front primeiro, pq já serve de feedback para o cliente e meio que evitava
alguns "esquecimentos" de funcionalidades, hahaha

Em qui, 5 de abr de 2018 11:42, luizbills <notifications@github.com>
escreveu:

**Comment by yanmagale** [comment]
Após o levantamento de requisitos, eu avaliaria o quão dependente o seu front-end está da API de Back-end. Faria um alinhamento dos endpoints e os dados que seriam retornados, e o desenvolvimento de ambos poderia seguir juntos.

Caso você dependa muito das ações de back-end, recomendo faze-lo primeiramente.

**Comment by mahenrique94** [comment]
Acho melhor começar pelo banco de dados, feito a elaboração e criação partir para o back e depois do back para o front, pois um depende do outro para estar pronto, assim, evita ter que ficar mockando dados.
